### ISSUE TITLE
[AvoidAnonymousTypes] False positive on UserAssignedIdentity generated from TypeSpec

### ISSUE THREAD

**Issue by mikeharder** [Azure/azure-openapi-validator#747]
## Problem
Rule `AvoidAnonymousTypes` raises a false-positive error on `UserAssignedIdentity` generated from TypeSpec.

### TypeSpec
```
  ...ManagedServiceIdentityProperty;
```

### Swagger
```
"Azure.ResourceManager.CommonTypes.ManagedServiceIdentityUpdate": {
  "type": "object",
  "description": "Managed service identity (system assigned and/or user assigned identities)",
  "properties": {
    "type": {
      "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentityType",
      "description": "The type of managed identity assigned to this resource."
    },
    "userAssignedIdentities": {
      "type": "object",
      "description": "The identities assigned to this resource by the user.",
      "additionalProperties": {
        "allOf": [
          {
            "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/UserAssignedIdentity"
          }
        ],
        "type": "object",
        "x-nullable": true
      }
    }
  }
},
```

## Workaround
Suppress in `readme.md` until root cause is addressed

## Example
https://github.com/Azure/azure-rest-api-specs-pr/pull/19710/checks?check_run_id=30708435029

## Related
https://github.com/Azure/typespec-azure/issues/1058.

**Comment by allenjzhang** [comment]
One more repo: https://github.com/Azure/azure-rest-api-specs/pull/32486/checks?check_run_id=36778866457

**Comment by allenjzhang** [comment]
This is affecting new Liftr specs which looked sqeaky clean from TypeSpec perspective. As the mandate of using ManagedIdentities, more TypeSpec will get flagged.
